### PR TITLE
admin/simpler-ci-and-build-infra

### PR DESCRIPTION
**Pull request by JacksonMaxfield**
### Description of Changes

_Include a description of the proposed changes._

Simplifies the CI / build infrastructure. Everything is now routed through `tox` so GitHub actions now just call tox instead of duplicating commands. Also switches to pre-commit for linting. This should make the process of contribution easier because someone can run `tox -e lint` and have it attempt to fix most mistakes for them, then run it again and ensure that mistakes _were_ fixed.

### CODE REVIEW

File: .github/workflows/build-main.yml [hunk 1]
```diff
@@ -70,19 +69,9 @@ jobs:
...
-        isort cdp_backend --check-only
-    - name: Check typing with mypy
-      run: |
-        mypy cdp_backend
-    - name: Check with black
-      run: |
-        black --check cdp_backend
+        pip install pre-commit
+    - name: Lint
+      run: pre-commit run --all-files --show-diff-on-failure
```
Comment (isaacna) [inline .github/workflows/build-main.yml:74]: Just checking, but does this run on everything in `cdp_backend` or only `cdp_backend/cdp_backend`?

**Review by isaacna — APPROVED** [review]
Looks good! Just had one question about how what directory level pre-commit runs on

File: .github/workflows/build-main.yml (hunk 1 above)
Comment (JacksonMaxfield) [reply .github/workflows/build-main.yml:74]: Good question. I think it runs on the whole repo but thats personally fine by me.

Comment (isaacna) [reply .github/workflows/build-main.yml:74]: Gotcha, my only comment would be sometimes I store things in the `cpd_backend` directory like files I'm going to run the pipeline on or various config files which could get in the way. Or maybe is there a way to only check files/changes that are added to git?

Comment (JacksonMaxfield) [reply .github/workflows/build-main.yml:74]: It should only be running lint and stuff on python files. Configs and such are ignored by black / flake8

Comment (JacksonMaxfield) [reply .github/workflows/build-main.yml:74]: Going to merge but if you encounter issues with this behaviour in the future let me know and I will figure out how to specify this to just the `cdp-backend/cdp_backend` dir

**Pull request merged**